### DISCUSSION TITLE
Use xfrm interface's public IPv6 address in pool?

### DISCUSSION THREAD

**Discussion by hgl** [strongswan/strongswan#1685] [Q&A] [answered]
I use an xfrm interface, and it's automatically assigned a public /64 IPv6 address by the system‘s dhcpv6 daemon. I wonder if there is a way to ask charon to read this /64 address from the interface and use it in a pool, so that my road warriors could also have public IPv6 addresses in the same subnet? Should I use the dhcp plugin, and does it support forwarding to a dhcpv6 server?

**Comment by tobiasbrunner** [comment, marked as answer]
> I wonder if there is a way to ask charon to read this /64 address from the interface and use it in a pool, so that my road warriors could also have public IPv6 addresses in the same subnet? 

There is no such automatism and since there is currently no IPv6 equivalent to the _farp_ plugin that would send NDP messages for roadwarrior clients that wouldn't work directly anyway (see [this old ticket](https://wiki.strongswan.org/issues/1008) for hints regarding NDP proxying via `ip -6 neigh`). I guess you could configure and load a pool dynamically via some DHCP/NM trigger.

> Should I use the dhcp plugin, and does it support forwarding to a dhcpv6 server?

No, the _dhcp_ plugin only supports IPv4.

**Comment by hgl** [comment]
Thanks for the quick reply, Tobias.

> I guess you could configure and load a pool dynamically via some DHCP/NM trigger.

By dynamic, do you mean load it via vici?

**Reply by tobiasbrunner** [reply]
Yep

**Reply by hgl** [reply]
> since there is currently no IPv6 equivalent to the farp plugin that would send NDP messages for roadwarrior clients

Sorry, I just realized I actually don't really understand this.

I wonder why I can ping a road warrior's IPv4 address from another LAN machine without the farp plugin? The road warrior's virtual IP is assigned from charon's pool. Running `ip neigh` on the server running charon shows no entry for the road warrior. Does the linux kernel provide access to road warriors directly without the need for arp? If so, I wonder exactly when is the farp plugin needed?

**Reply by tobiasbrunner** [reply]
In your case it's not really an issue, as you intend to assign IPs from the host's `/64` address. It's more a general problem if roadwarriors should be in the same subnet as the server (see [here](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_hosts_on_the_lan)).

**Reply by hgl** [reply]
It starts to make sense. Thanks.

Do you think it's actually a good idea to give public IPv6 to road warriors? I found out that they also work (i.e., accessing LAN machines the internet via the server) if I just specify a random ULA in the pool. Does giving public IPv6 allow them, for example, listen on the public IPv6 and provide an http server that the internet can visit via the gateway server?

**Reply by tobiasbrunner** [reply]
> I found out that they also work (i.e., accessing LAN machines the internet via the server) if I just specify a random ULA in the pool.

That requires a NAT, which can have its own issues.

> Does giving public IPv6 allow them, for example, listen on the public IPv6 and provide an http server that the internet can visit via the gateway server?

That depends on the firewall rules on the server.
